cmds/core/ls/ls_linux_test.go

```diff
@@ -23,26 +25,54 @@ import (
 //     = mkdev(0x456, 0x12378)
 //     = (0x12378 & 0xff) | (0x456 << 8) | ((0x12378 & ~0xff) << 12)
 //     = 0x12345678
-func TestLargeDevNumber(t *testing.T) {
-	if uid := os.Getuid(); uid != 0 {
-		t.Skipf("test requires root, your uid is %d", uid)
-	}
-
-	// Make the node.
-	tmpDir := t.TempDir()
-	file := filepath.Join(tmpDir, "large_node")
-	if err := unix.Mknod(file, 0o660|unix.S_IFBLK, 0x12345678); err != nil {
-		t.Fatal(err)
-	}
-	defer os.Remove(file)
 
-	// Run "ls -l large_node".
-	out, err := testutil.Command(t, "-l", file).Output()
+// Test listName func
+func TestListNameLinux(t *testing.T) {
+	testutil.SkipIfNotRoot(t)
+	// Create a directory
+	d, err := os.MkdirTemp(os.TempDir(), "li.st")
 	if err != nil {
-		t.Fatal(err)
+		t.Errorf("Failed to create tmp dir: %v", err)
+	}
+	if err := unix.Mknod(filepath.Join(d, "large_node"), 0o660|unix.S_IFBLK, 0x12345678); err != nil {
+		t.Error(err)
 	}
-	expected := regexp.MustCompile(`^\S+ \S+ \S+ 1110, 74616`)
-	if !expected.Match(out) {
-		t.Fatal("expected device number (1110, 74616), got:\n" + string(out))
+	defer os.RemoveAll(d)
+
+	// Creating test table
+	for _, tt := range []struct {
```

Comment:
No need to have a table-driven test if there is only 1 test.

Reply:
If it is table-driven, regardless if it is 1 test, it can be easily expanded with more tests it needed.

Reply:
I think it is no disadvantage if it stays table-driven.

Reply:
Imo, it's more code than there needs to be. It's not idiomatic Go to come up with generalizations before they're absolutely necessary - unless you can think of other test cases, a non-table-test would be preferred.